pkg/skaffold/runner/listen.go

```diff
@@ -60,29 +62,45 @@ func (l *SkaffoldListener) WatchForChanges(ctx context.Context, out io.Writer, d
 
 	l.LogWatchToUser(out)
 
+	eg, egCtx := errgroup.WithContext(ctx)
+	devCtx, cancelDev := context.WithCancel(egCtx)
 	for {
 		select {
-		case <-ctx.Done():
+		case <-egCtx.Done():
```

Comment:
Since you don't actually call `eg.Go()` on the first pass, won't you immediately hit this case, call `cancelDev()`, and then return nil?

Reply:
I can reply since I've taken over this PR.

The `context` is valid right from the initialization of the `eg` struct so waiting on the `Done` channel should work fine. `eg.Go` actually sets up blocking `eg.Wait`. So calling `eg.Wait` before `eg.Go` would return immediately and close this `Done` channel but that doesn't happen here. 
You can look at the source [here](https://github.com/golang/sync/blob/43a5402ce75a/errgroup/errgroup.go#L32)